### ISSUE TITLE
deprecate @storybook/vue in 7.6

### ISSUE THREAD

**Issue by vanessayuenn** [storybookjs/storybook#24779]
Vue2 reaches EOL on 2023-12-31, and we will remove support for it in Storybook in 8.0. Ahead of that, we should deprecate @storybook/vue in 7.6

#### Tasks
- [x] write up rationale for deprecating @storybook/vue as an RFC in GitHub discussion 
- [ ] deprecate @storybook/vue in 7.6 and point to 👆 the discussion

**Comment by vanessayuenn** [comment]
@shilman to decide on whether to deprecate package or to warn user (and how)

**Comment by shilman** [comment]
Let's deprecate `@storybook/vue` in 7.6. This is the loudest signal of what's to come. The deprecation warning should point to a GH discussion where we explain the rationale for removing Vue2 support in 8.0. This also gives us a chance to reverse the decision before 8.0 if we're missing something.

**Comment by vanessayuenn** [comment]
alright i updated the issue to capture the actual work needed. 

@shilman will you write the rationale RFC? Sounds like we cannot do the deprecation without the link to point it to. @yannbf I supposed it makes sense to assign it to the next empathy person, in this case @ndelangen.

**Comment by yannbf** [comment]
Here's the link to the RFC: https://github.com/storybookjs/storybook/discussions/24951

cc @JReinhold (currently on empathy)

**Comment by vanessayuenn** [comment]
> * At the end of the publishing workflow, deprecate the packages again, so that when 7.6.Y is released, the packages are deprecated again.
> * [ ]  Get 👆 on `main` as well.

@JReinhold So we need to keep doing this until 8.0 is out of prerelease, right?

**Comment by JReinhold** [comment]
@vanessayuenn yes unless we add it as a script at the end of every publish. That was my original intent but we never got that far
